pkgs/os-specific/linux/yt6801/default.nix

```
@@ -0,0 +1,49 @@
+{
+  stdenv,
+  lib,
+  fetchzip,
+  kmod,
+  kernel,
+}:
+
+let
+  modDestDir = "$(out)/lib/modules/${kernel.modDirVersion}/kernel/drivers/net/ethernet/motorcomm";
+
+in
+stdenv.mkDerivation rec {
+  pname = "yt6801";
+
+  version = "1.0.29";
+  src = fetchzip {
+    stripRoot = false;
+    url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/20240812/yt6801-linux-driver-${version}.zip";
+    sha256 = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
```

Also: are you sure that the date in the download link is the same for all versions? If not then it should be its own updatable variable
```suggestion
    url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/20240812/yt6801-linux-driver-${finalAttrs.version}.zip";
    hash = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
    stripRoot = false;
```